### ISSUE TITLE
Intermittent raycast bug in Unity 5.5

### ISSUE THREAD

**Issue by paulhayes** [wacki/Unity-VRInputModule#5]
I'm getting an odd issue in Unity 5.5 ( in f3 and latest p4 ). Often the Canvas is not picked and the indicator graphical ray just shoots right though it. Stopping and Starting playback in the editor can fix/break it.

**Comment by paulhayes** [comment]
Think I've found the issue. Seems to be Editor related. 

https://forum.unity3d.com/threads/baseinputmodule-broken-in-5-5.449206/

**Comment by JashanChittesh** [comment]
It took me a while to figure this one out but I finally found the issue: Some very smart person added OnApplicationFocus to the Unity UI EventSystem in 5.5, and this will pause the EventSystem when the application loses focus. That will result in Process no longer be called on our custom in-VR input modules.

I believe the only solution is to write our own implementations of EventSystem that don't to stupid things like that ;-)

**Comment by paulhayes** [comment]
Yeah that's a stinker. On one level it makes sense that they put that in their input code, but obviously it makes no sense for VR. 

Another solution is we could listen for when a headset it donned ( At lease for the Oculus ), and use that to get application focus.

**Comment by andybak** [comment]
This explains some problems I had playing with the VRTK examples.

Has this issue been logged with Unity or is that a fool's errand?

**Comment by andybak** [comment]
Ignore me. I've spotted the forum thread mentioned above. I posted there asking if there's a public issue we can get updates for.

**Comment by tkeene** [comment]
Thanks everybody, this thread was very helpful in solving this issue.

https://www.reddit.com/r/Unity3D/comments/699rej/custom_pointerinputmodule_issues_when_window/

    using UnityEngine.EventSystems;
    
    public class CustomCanvasEventSystem : EventSystem
    {
        protected override void OnApplicationFocus(bool hasFocus)
        {
            // Workaround because the default EventSystem implementation stops polling pointer input in our PointerInputModule if the application loses focus (such as when I use the desktop viewer to click on another application's window).
            base.OnApplicationFocus(true);
        }
    }

This is a slightly annoying workaround. It would be nice if EventSystem had a checkbox or enum field for changing this behavior.
